sklearn/linear_model/logistic.py

```diff
@@ -1561,64 +1556,67 @@ def fit(self, X, y, sample_weight=None):
 
         X, y = check_X_y(X, y, accept_sparse='csr', dtype=np.float64,
                          order="C")
-
+        check_classification_targets(y)
         if self.solver == 'sag':
             max_squared_sum = row_norms(X, squared=True).max()
         else:
             max_squared_sum = None
 
-        check_classification_targets(y)
+        # Encode for string labels
```

Comment:
`classes_labels` was done to match the naming of `encoded_labels`. I'll refactor it back to `classes` if it seems clearer...